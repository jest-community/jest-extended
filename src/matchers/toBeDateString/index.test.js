import matcher from './';

expect.extend(matcher);

describe('.toBeDateString', () => {
  test('passes when given a date string', () => {
    expect(new Date().toISOString()).toBeDateString();
  });

  test('fails when not given a date string', () => {
    expect(() => expect('not a date').toBeDateString()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeDateString', () => {
  test('passes when not given a date string', () => {
    expect('not a date').not.toBeDateString();
  });

  test('fails when given a date string', () => {
    expect(() => expect('2018-01-01T13:00:00.000Z').not.toBeDateString()).toThrowErrorMatchingSnapshot();
  });
});
