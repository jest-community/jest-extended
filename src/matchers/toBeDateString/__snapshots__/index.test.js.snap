// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeDateString fails when given a date string 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeDateString()</>

Expected value to not be a date string received:
  <red>\\"2018-01-01T13:00:00.000Z\\"</>"
`;

exports[`.toBeDateString fails when not given a date string 1`] = `
"<dim>expect(</><red>received</><dim>).toBeDateString()</>

Expected value to be a date string received:
  <red>\\"not a date\\"</>"
`;
